rest/tests/test_vocabulary.py

```diff
@@ -896,6 +896,134 @@ def test_vocab_num_queries(self):
+
+        # Facet counts should not change
+        self.assertEqual(
+            self.get_results()['facet_counts'][vocab_key]['facet']['label'],
+            "brand new name"
+        )
+        self.assertEqual(
+            self.get_results()['facet_counts'][vocab_key]['values'],
+            [{'count': 1, 'key': str(term_id), 'label': term_label}]
+        )
+        self.assertEqual(
+            self.get_results(selected_facets=["{v}_exact:{t}".format(
+                v=vocab_key,
+                t=term_id
+            )])['count'],
+            1
+        )
+
+    def test_term_rename(self):
+        """Test that index updates properly after a term rename."""
+        vocab_result = self.create_vocabulary(self.repo.slug)
+        vocab_key = make_vocab_key(vocab_result['id'])
+        vocab_slug = vocab_result['slug']
+        term_result = self.create_term(self.repo.slug, vocab_slug)
+        term_id = term_result['id']
+        term_slug = term_result['slug']
+
+        # Update vocabulary for resource type, assign term
+        self.patch_vocabulary(self.repo.slug, vocab_slug, {
+            "learning_resource_types": [
+                self.resource.learning_resource_type.name
+            ]
+        })
+        self.patch_learning_resource(self.repo.slug, self.resource.id, {
+            "terms": [term_slug]
+        })
+
+        self.assertEqual(
+            self.get_results()['facet_counts'][vocab_key]['values'],
+            [{'count': 1, 'key': str(term_id), 'label': term_result['label']}]
+        )
+        self.assertEqual(
+            self.get_results(selected_facets=["{v}_exact:{t}".format(
+                v=vocab_key,
+                t=term_id
+            )])['count'],
+            1
+        )
+
+        # Rename term
+        self.patch_term(self.repo.slug, vocab_slug, term_slug, {
+            "label": "brand new label"
+        })
+
+        # Facet counts should not change
+        self.assertEqual(
+            self.get_results()['facet_counts'][vocab_key]['values'],
+            [{'count': 1, 'key': str(term_id), 'label': "brand new label"}]
+        )
+        self.assertEqual(
+            self.get_results(selected_facets=["{v}_exact:{t}".format(
+                v=vocab_key,
+                t=term_id
+            )])['count'],
+            1
+        )
+
+    def test_empty_vocab_facet_count(self):
+        """Test that we get vocabulary label and not something else."""
+        vocab_dict = dict(self.DEFAULT_VOCAB_DICT)
+        vocab_dict['name'] = 'name with spaces'
+        vocab_result = self.create_vocabulary(self.repo.slug, vocab_dict)
+        vocab_slug = vocab_result['slug']
+        vocab_id = vocab_result['id']
+        vocab_key = make_vocab_key(vocab_id)
+
+        self.assertEqual(
+            self.get_results()['facet_counts'][vocab_key]['facet']['label'],
+            'name with spaces'
+        )
+
+        self.create_term(self.repo.slug, vocab_slug)
+        self.assertEqual(
```

Comment:
There was a bug where if a vocab had no terms the vocab slug would show up in place of the vocab label (#779)